### PR TITLE
Add WebWorker Support

### PR DESCRIPTION
**Pull request by jteplitz**
The app is fully functional with both Pre Rendering and WebWorkers, but preboot does not yet work with WebWorkers (#31). I'm planning to design a solution to this tomorrow.

### CODE REVIEW

**Comment by jeffbcross** [comment]
Let's review this and rebase against Rob's pending PR #29 this morning. I will also submit a separate PR to lock down our npm dependency versions.

**Comment by jteplitz** [comment]
I fixed the preboot issues. However, we still need to depend on a setTimeout hack at the moment. Look at #31 for details.

This also doesn't work with firebase since zones are not properly patching WebSockets in a WebWorker.

File: src/main-server.ts [hunk 1]
```diff
@@ -2,7 +2,11 @@
 
 import * as path from 'path';
 import * as express from 'express';
-import {SERVER_LOCATION_PROVIDERS, ng2engine} from 'angular2-universal-preview/dist/server';
+import {
+  SERVER_LOCATION_PROVIDERS,
+  ng2engine,
+  REQUEST_URL
+} from 'angular2-universal-preview/dist/server';
```
Comment (PatrickJS) [inline src/main-server.ts:9]: you can simply use `angular2-universal-preview`

**Comment by jeffbcross** [comment]
LGTM, but I will leave it to @robwormald to merge so he can figure out how to reconcile it with his own stuff.

**Comment by PatrickJS** [comment]
I can also include an async-preboot-complete to universal if you want the code to look more clean (after merging this of course)

**Comment by jteplitz** [comment]
@gdi2290 What would that look like?

**Comment by PatrickJS** [comment]
nothing fancy, something like [`prebootComplete` modules/universal/browser/src/bootstrap.ts#L20...L25](https://github.com/angular/universal/blob/295d3bca40e1c05fc206dad11b91dc923141ba9f/modules/universal/browser/src/bootstrap.ts#L20...L25) all it would do is emit `complete` on the next tick.

**Comment by jteplitz602** [comment]
@gdi2290

**Comment by jteplitz** [comment]
@gdi2290 No, I think that would just be a bandaid over the real problem. The bootstrap promise should not resolve until after the root component has been rendered. I'll take a look at that next week.

**Comment by jeffbcross** [comment]
I'm going to attempt rebasing this and merging now that other PRs have been merged...

**Comment by jeffbcross** [comment]
Merged at ade67cf9b1acb6eefc4b7b718167fe6a5fce4a65

**Pull request closed without merging**